### PR TITLE
Use new che:theia generator

### PR DESCRIPTION
**Pull request by evidolob**
This pull use new version of `che:theia` generator and own `extensions.yml` for extensions

### CODE REVIEW

File: dockerfiles/theia/build.sh [hunk 1]
```diff
@@ -10,6 +10,19 @@
...
+
+if [ -d "${LOCAL_ASSEMBLY_DIR}" ]; then
+  rm -r "${LOCAL_ASSEMBLY_DIR}"
+fi
+
+#in mac os 'cp' cannot create destination dir, so create it first
+mkdir ${LOCAL_ASSEMBLY_DIR}
+
+echo "Copying ${base_dir}/../../extensions/extensions.yml --> ${LOCAL_ASSEMBLY_DIR}/extensions.yml"
+cp "${base_dir}/../../extensions/extensions.yml" "${LOCAL_ASSEMBLY_DIR}/"
```
Comment (benoitf) [inline dockerfiles/theia/build.sh]: I would have used ln -s to just create symlink (and avoid copy each time) but I'm fine with the cp

Comment (evidolob) [reply dockerfiles/theia/build.sh]: Docker cannot follow symlinks https://github.com/moby/moby/issues/1676

Comment (benoitf) [reply dockerfiles/theia/build.sh]: Hard link then (like we did there two years ago for che-server)
https://github.com/eclipse/che/blob/5.1.0/dockerfiles/che/build.sh#L27

Comment (benoitf) [reply dockerfiles/theia/build.sh]: but here the file size is so small that it's quite useless

Comment (evidolob) [reply dockerfiles/theia/build.sh]: Even now we use `cp` https://github.com/eclipse/che/blob/master/dockerfiles/che/build.sh#L31 )

Comment (benoitf) [reply dockerfiles/theia/build.sh]: the reason is different. (to have moved from ln to cp) It was to do some very weird stuff

**Pull request merged**